### ISSUE TITLE
Mxnet has very bad performance on iris dataset and sonar dataset comparing Tensorflow

### ISSUE THREAD

**Issue by qinhui99** [apache/incubator-mxnet#5682]
Not only in train Accuracy, but also in prediction.  I test some codes both in mxnet and in Tensorflow. Then I got the following result:
1. sonar dataset:
a)Tensorflow codes :
train Accuracy: 0.857143
prediction:  I test three samples.
The original label is  [0,1,0], the prediction is [1,1,0]. 
About 66% accuracy in prediction.

b) mxnet codes:
train Accuracy: 0.6
prediction:   I test three samples.
The original label is  [0,1,0], the prediction is [1,1,1].  
About 33% accuracy in prediction.

2. iris dataset:
a)Tensorflow codes :
train Accuracy: 1.0
prediction:  I test ten samples .
The original label is  [1,0,2,1,0,1,1,1,2,1], 
the prediction is       [1,0,2,1,0,0,1,2,2,1] .  
About 80% accuracy in prediction.

b) mxnet codes:
train Accuracy: 0.97
prediction:  I test ten samples.
The original label is [1,0,2,1,0,1,1,1,2,1], 
the prediction is      [2,1,2,2,1,2,2,2,2,2].
About 20% accuracy in prediction. 

As you can see the above-mentioned,   when I use model to do prediction , mxnet codes always get the worst result than Tensorflow codes. I don't know why.  I hope the mxnet developers should pay attention to these.  Could your guys write some codes on iris dataset or sonar dataset to show how mxnet work ? 

My codes can be download at https://pan.baidu.com/s/1jHZoMZO. You can run them to compare mxnet and tensorflow.

**Comment by cwgreene** [comment]
Why not use a gist or your own github account for your code? The link is dead.

**Comment by qinhui99** [comment]
My codes can be downloaded from [](https://github.com/qinhui99/mxnetSamples).

**Comment by qinhui99** [comment]
Sorry.  I submited the codes in github. You can see them in https://github.com/qinhui99/mxnetSamples.git.


________________________________
发件人: Chris Greene <notifications@github.com>
发送时间: 2017年4月10日 6:41
收件人: dmlc/mxnet
抄送: hui qin; Author
主题: Re: [dmlc/mxnet] Mxnet has very bad performance on iris dataset and sonar dataset comparing Tensorflow (#5682)


Why not use a gist or your own github account for your code? The link is dead.

―

**Comment by kishoreabhishek** [comment]
Try  changing the number of epochs to 50

**Comment by qinhui99** [comment]
Thanks for your reply. But it is no use to increase the epoch number. The results do not change much.

**Comment by yajiedesign** [comment]
you forget normalization with `new_samples`

**Comment by orchidmajumder** [comment]
I have worked on your notebook and changed small parameters and I am getting 1.0 train accuracy and 0.96 validation accuracy. Here is my notebook: https://gist.github.com/orchidmajumder/fe6d1cfc657010e57f5e1ddf67ffbe02

**Comment by orchidmajumder** [comment]
And here is another example with 60-20-20 train validation test split: https://gist.github.com/orchidmajumder/81c3b4ce9d70dd5e1e08b1d3bce50ea9

train accuracy: 1.0, validation: .93, test: .96

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
